packages/cli/src/utils/api.js

```diff
@@ -364,6 +366,61 @@ const validateApp = async (definition) => {
   });
 };
 
+// TODO: docstring, fix ensureDir failure
+const downloadSourceZip = async () => {
+  const linkedAppConfig = await getLinkedAppConfig(undefined, false);
+  if (!linkedAppConfig.id) {
+    throw new Error(
+      `This project hasn't yet been associated with an existing Zapier integration.\n\nIf it's a brand new integration, run \`${colors.cyan(
+        'zapier register'
+      )}\`.\n\nIf this project already exists in your Zapier account, run \`${colors.cyan(
+        'zapier link'
+      )}\` instead.`
+    );
+  }
+
+  const credentials = await readCredentials(true);
+
+  const url = constants.ENDPOINT + `/apps/${linkedAppConfig.id}/latest/pull`
+
+  const requestOptions = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json; charset=utf-8',
+      'User-Agent': `${constants.PACKAGE_NAME}/${constants.PACKAGE_VERSION}`,
+      'X-Requested-With': 'XMLHttpRequest',
+      'X-Deploy-Key' : credentials[constants.AUTH_KEY]
+    },
+  };
+  const response = await fetch(url, requestOptions);
```

Comment:
There's a `callAPI()` function we can use to make API requests. So we don't need to deal with authentication or headers here. If `callAPI()` can't be used to download a binary file, feel free to change it so it supports that.

It's also a bit strange that we're expecting a zip file while the headers include things like `Accept: 'application/json'` and `X-Requested-With: 'XMLHttpRequest'`.